### PR TITLE
Unit testing for notification functionality

### PR DESCRIPTION
**Pull request by murradkh**

### CODE REVIEW

File: myapp/admin.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from django.contrib import admin
 from .models import Stock, Profile, Notification, WatchedStock
-from .sub_models.notification_rules import ChangeStatusRule, ChangeThresholdRule, PriceThresholdRule, \
+from .sub_models.notification_rules_models import ChangeStatusRule, ChangeThresholdRule, PriceThresholdRule, \
```
Comment (kamalq97) [inline myapp/admin.py:3]: This will definitely create conflicts 😅

File: myapp/scheduler/tests/test_notification_handler_updater.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+
+        rule = RecommendationAnalystRule.objects.create(watched_stock=watched_stock, category='S',
+                                                        threshold_recommenders_percentage=34, fired=False)
+        dump_data = {'ratingBuy': 1, 'ratingOverweight': 1, 'ratingHold': 1, 'ratingUnderweight': 1, 'ratingSell': 2,
+                     'ratingNone': 0, 'ratingScaleMark': 1, 'consensusStartDate': 1648034225850,
+                     'consensusEndDate': None, 'corporateActionsAppliedDate': 1533502145053}
+
+        mocked_stock_api.get_analyst_recommendations.return_value = dump_data
+        recommendation_analyst_rule()
+        self.assertEqual(len(Notification.objects.filter(pk=8)), 0)
```
Comment (kamalq97) [inline myapp/scheduler/tests/test_notification_handler_updater.py:411]: Sooooo.... much code! I must admit I read most of it but not all

File: myapp/scheduler/tests/test_stock_api_update.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+
+class JbsTestCase(TestCase):
+
+    def test_api_update(self):
+        stock_api_update()
+        value_1 = Stock.objects.filter(top_rank=1)[0].last_modified
+        stock_api_update()
+        value_2 = Stock.objects.filter(top_rank=1)[0].last_modified
+        self.assertNotEqual(value_1, value_2)
```
Comment (kamalq97) [inline myapp/scheduler/tests/test_stock_api_update.py:14]: Did not Malak write a similar test some time ago (?)

**Review by kamalq97 — APPROVED** [review]
Lots of code changes but the unit tests are great and seem to cover edge cases very well! 👍

File: myapp/scheduler/tests/test_stock_api_update.py (hunk 3 above)
Comment (murradkh) [reply myapp/scheduler/tests/test_stock_api_update.py:14]: no
